FinSearchUnified/BusinessLogic/ExportService.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+namespace FinSearchUnified\BusinessLogic;
+
+use Doctrine\ORM\EntityNotFoundException;
+use Exception;
+use FINDOLOGIC\Export\Data\Item;
+use FINDOLOGIC\Export\Helpers\EmptyValueNotAllowedException;
+use Shopware\Models\Article\Article;
+use Shopware\Models\Category\Category;
+use Shopware\Models\Customer\Customer;
+
+class ExportService
+{
+    /**
+     * @var \Shopware\Models\Customer\Repository
+     */
+    protected $customerRepository;
+
+    /**
+     * @var \Shopware\Models\Article\Repository
+     */
+    protected $articleRepository;
+
+    /**
+     * @var FindologicArticleFactory
+     */
+    protected $findologicArticleFactory;
+
+    /**
+     * @var string
+     */
+    public $shopKey;
+
+    /**
+     * @var Category
+     */
+    protected $baseCategory;
+
+    /**
+     * @var array
+     */
+    protected $allUserGroups;
+
+    /**
+     * @var array[]
+     */
+    protected $errors = [];
+
+    public function __construct($shopkey, Category $baseCategory)
+    {
+        $this->customerRepository = Shopware()->Container()->get('models')->getRepository(Customer::class);
+        $this->articleRepository = Shopware()->Container()->get('models')->getRepository(Article::class);
+        $this->findologicArticleFactory = Shopware()->Container()->get('fin_search_unified.article_model_factory');
+        $this->shopKey = $shopkey;
+        $this->baseCategory = $baseCategory;
+        $this->allUserGroups = $this->customerRepository->getCustomerGroupsQuery()->getResult();
+    }
+
+    public function fetchTotalProductCount()
+    {
+        $countQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('count(articles.id)')
+            ->where('articles.active = :active')
+            ->orderBy('articles.id')
+            ->setParameter('active', true);
+
+        return $countQuery->getQuery()->getScalarResult()[0][1];
+    }
+
+    public function fetchAllProducts($start, $count)
+    {
+        $articlesQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('articles')
+            ->where('articles.active = :active')
+            ->orderBy('articles.id')
+            ->setParameter('active', true);
+
+        if ($count > 0) {
+            $articlesQuery->setMaxResults($count)->setFirstResult($start);
+        }
+
+        return $articlesQuery->getQuery()->execute();
+    }
+
+    public function fetchProductById($productId)
+    {
+         $articlesQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('articles')
+            ->where('articles.id = :productId')
+            ->orWhere('articles.supplier = :productId')
+            ->setParameter('productId', $productId);
+
+        $shopwareArticles = $articlesQuery->getQuery()->execute();
+
+        if (count($shopwareArticles) === 0) {
+            $this->errors['general'][] = 'No article found with given ID';
```

Comment:
I know it sounds silly, but what if a product id is "general"? How could you differentiate between the errors? We should unify the structure for the JSON for SW5 and SW6. How about this?

```json
{
    "errors": {
        "general": [
            "Why are you even trying?"
        ],
        "products": [
            {
                "id": "1234",
                "errors": [
                    "Not enough blubbergurken."
                ]
            },
            {
                "id": "abcd-efgh",
                "errors": [
                    "OOF. Something went wrong!",
                    "OOF. Something else went also wrong!",
                ]
            }
        ]
    }
}
```

If you have any other suggestions, or if you think that format is incomplete, please let me know.

Reply:
Please check it out once more

![image](https://user-images.githubusercontent.com/51320851/92747419-786f1580-f384-11ea-9358-d2af7637d71e.png)

![image](https://user-images.githubusercontent.com/51320851/92747494-89b82200-f384-11ea-8b94-463b62223cf2.png)

Reply:
@TobiasGraml11 the responses you have posted should look like that:

*1st response*
```json
{
    "errors": {
        "general": [
            "No article found for id 20"
        ],
        "products": []
    }
}
```

*2nd response*
```json
{
    "errors": {
        "general": [],
        "products": [
            {
                "id": 2,
                "errors": [
                    "Product is not active.",
                    "Main Detail is not active or not available.",
                    "shouldBeExported is false."
                ]
            },
            {
                "id": 10,
                "errors": [
                    "All configured categories are inactive.",
                    "shouldBeExported is false.",
                ]
            },
            {
                "id": 13,
                "errors": [
                    "All configured categories are inactive.",
                    "shouldBeExported is false.",
                ]
            }
        ]
    }
}
```

Other than that it looks really nice!